Assets/Scripts/Marker.cs

```diff
@@ -0,0 +1,82 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class Marker : MonoBehaviour {
+
+  [SerializeField] private Sprite icon;
+  [SerializeField] private float iconBoxSize;
+
+  private const float markerDuration = 5f;
+
+  private static RectTransform uiRoot = null;
+  private static Camera mainCamera = null;
+
+  public void Activate() {
+    if (uiRoot == null) {
+      uiRoot = UIEventListener.Instance.transform as RectTransform;
+    }
+    GameObject markerObject = new("UIMarker for: " + name, typeof(Image));
+    markerObject.transform.SetParent(uiRoot, false);
+    StartCoroutine(DisplayMarker(markerObject.transform as RectTransform));
+  }
+
+  private IEnumerator DisplayMarker(RectTransform uiMarker) {
+    if (mainCamera == null) {
+      mainCamera = Camera.main;
+    }
+    uiMarker.GetComponent<Image>().sprite = icon;
+    uiMarker.sizeDelta = new(iconBoxSize, iconBoxSize);
+    //uiMarker.sizeDelta = icon.textureRect.max;
+    float markerTimer = markerDuration;
+    float edgeBuffer = uiMarker.rect.size.x / 2;
+    while (markerTimer >= 0) {
+      var screenPoint = WorldToScreenPointProjected(mainCamera, transform.position);
+      uiMarker.position = ScreenPointEdgeClamp(screenPoint, edgeBuffer);
+
+      markerTimer -= Time.deltaTime;
+
+      // This is a pretty cruddy fix for hiding these on game pause, but it's better than nothing for now (for a better implementation, UIEventListener should probably hide these on pause)
+      if (Time.deltaTime == 0f) {
+        uiMarker.gameObject.SetActive(false);
+      } else {
+        uiMarker.gameObject.SetActive(true);
+      }
+
+      yield return null;
+    }
+    Destroy(uiMarker.gameObject);
```

Comment:
Instead of destroying the gameObject, and then recreating it (for possible later events) why don't we just set enabled to false, and then true on activate? That way we can also just do the initializing done in lines 20, 29 and 30 once in Start() and not worry about it again.